crates/types/src/services/txpool.rs

```diff
@@ -333,7 +334,7 @@ impl From<&PoolTransaction> for CheckedTransaction {
 /// The status of the transaction during its life from the tx pool until the block.
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub enum TransactionStatus {
+pub enum TransactionStatusStorage {
```

Comment:
It would be nice to use some other name, since we also use this type as a result of execution, maybe `TransactionExecutionStatus` , idk

Reply:
`TransactionExecutionStatus` already exists in `executor`: https://github.com/FuelLabs/fuel-core/blob/master/crates/types/src/services/executor.rs#L185-L190

We also seem to already have what we need (albeit there are differences in fields) in the form of `TransactionExecutionResult`: https://github.com/FuelLabs/fuel-core/blob/master/crates/types/src/services/executor.rs#L195-L218

For now I used the explicit qualification (`txpool::TransactionExecutionStatus`) to resolve conflicts to avoid introducing yet another name for transaction status.

Follow-up issue for grooming: https://github.com/FuelLabs/fuel-core/issues/2795